### PR TITLE
contact mananager polymorphism

### PR DESCRIPTION
**Pull request by nmm0**

### CODE REVIEW

**Comment by nmm0** [comment]
A few notes: In the interest in getting this merged faster I haven't fully ported over the Kokkos code to the separate KokkosContactManager class. Additionally, a new TODO would be configuration file selection of contact backend

File: src/contact/parallel/bvh_contact_manager.cc [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      std::vector<int> vtk_vertices(nodes.size());
+      std::vector< std::vector<int> > vtk_triangles(faces.size());
+
+      vis_file << "POINTS " << nodes.size() + 3*faces.size() << " float" << std::endl;
+
+      for (int i=0 ; i<nodes.size() ; i++) {
+        vis_file << nodes[i].coord_1_x_ << " " << nodes[i].coord_1_y_ << " " << nodes[i].coord_1_z_ << std::endl;
+        vtk_vertices[i] = i;
+      }
+      int offset = static_cast<int>(vtk_vertices.size());
```
Comment (japlews) [inline src/contact/parallel/bvh_contact_manager.cc:197]: For large models you could consider using the base64 format encoding provided by the VTK library itself.

**Comment by uhetmaniuk** [comment]
@nmm0 Does the merge close #11 ?

**Comment by nmm0** [comment]
@uhetmaniuk Yes, I'll close the issue

**Pull request merged**